### PR TITLE
Add password hashing

### PR DESCRIPTION
**Pull request by krzmaz**

### CODE REVIEW

File: src/webapi/src/controller/user_repository.rs [hunk 1]
```diff
@@ -17,14 +22,19 @@ pub async fn insert_user(
...
     INSERT INTO users (id, username, password, created_at)
     VALUES ($1, $2, $3, $4)
             "#,
         Uuid::new_v4(),
         user.username,
-        user.password.expose_secret(),
+        hashed_password,
         Utc::now()
     )
     .execute(pool)
```
Comment (mrozycki) [inline src/webapi/src/controller/user_repository.rs:40]: Case in point: how are you planning to verify the password later if you don't store the salt?

**Review by mrozycki — COMMENTED** [review]
I'm not sure how much value this has on its own, without also implementing a login mechanism.

File: src/webapi/src/controller/user_repository.rs (hunk 1 above)
Comment (krzmaz) [reply src/webapi/src/controller/user_repository.rs:40]: maybe the naming could be better, but the Pbkdf2 crate returns the password in PHC string format, which contains the salt. See https://docs.rs/pbkdf2/0.12.2/pbkdf2/

**Comment by mrozycki** [comment]
Also, while I'm no security expert myself, it seems like Argon2 is suggested for new projects: https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#peppering

**Review by mrozycki — APPROVED** [review]

**Pull request merged**